marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/MintTokenEvent.hs

```diff
@@ -0,0 +1,281 @@
+  let redeemer = findRedeemer ix
+  pure $
+    MintAsset
+      (fromMaryPolicyID policyId')
+      (fromMaryAssetName assetName)
+      (C.Quantity quantity)
+      (toAssetRedeemer <$> redeemer)
+
+fromMaryPolicyID :: PolicyID StandardCrypto -> C.PolicyId
+fromMaryPolicyID (PolicyID sh) = C.PolicyId (C.fromShelleyScriptHash sh) -- from cardano-api:src/Cardano/Api/Value.hs
+
+fromMaryAssetName :: AssetName -> C.AssetName
+fromMaryAssetName (AssetName n) = C.AssetName $ Short.fromShort n -- from cardano-api:src/Cardano/Api/Value.hs
+
+fromAlonzoData :: LA.Data ledgerera -> C.ScriptData
+fromAlonzoData = C.fromPlutusData . getPlutusData -- from cardano-api:src/Cardano/Api/ScriptData.hs
+
+mkSqliteIndexer :: FilePath -> IO MintTokenEventIndexer
+mkSqliteIndexer dbPath = do
+  c <- SQL.open dbPath
+
+  SQL.execute_ c "PRAGMA journal_mode=WAL"
+
+  SQL.execute_
+    c
+    [sql|CREATE TABLE IF NOT EXISTS
+          minting_policy_events
+          ( slotNo INT NOT NULL
+          , blockHeaderHash INT NOT NULL
+          , blockNo INT NOT NULL
+          , txId BLOB NOT NULL
+          , txIndexInBlock BLOB NOT NULL
+          , policyId BLOB NOT NULL
+          , assetName TEXT NOT NULL
+          , quantity INT NOT NULL
+          , redeemerHash BLOB
+          , redeemerData BLOB
+          )|]
+
+  SQL.execute_
+    c
+    [sql|CREATE INDEX IF NOT EXISTS
+         minting_policy_events__txId_policyId
+         ON minting_policy_events (txId, policyId)|]
+
+  createLastSyncTable c
+
+  let mintEventInsertQuery :: SQL.Query -- Utxo table SQL statement
+      mintEventInsertQuery =
+        [sql|INSERT
+               INTO minting_policy_events (
+                 slotNo,
+                 blockHeaderHash,
+                 blockNo,
+                 txIndexInBlock,
+                 txId,
+                 policyId,
+                 assetName,
+                 quantity,
+                 redeemerHash,
+                 redeemerData
+              ) VALUES
+              (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)|]
+
+  pure $
+    Core.SQLiteIndexer
+      c
+      [
+        [ Core.SQLInsertPlan timedMintEventsToTimedMintEventList mintEventInsertQuery
+        ]
+      ]
+      Core.genesis
+
+timedMintEventsToTimedMintEventList
+  :: Core.Timed point MintTokenEvents
+  -> [Core.Timed point MintTokenEvent]
+timedMintEventsToTimedMintEventList =
+  traverse $ toListOf $ mintTokenEventsAssets . folded
+
+instance (MonadIO m) => Core.Rollbackable m MintTokenEvents Core.SQLiteIndexer where
+  rollback C.ChainPointAtGenesis ix = do
+    let c = ix ^. Core.handle
+    liftIO $ SQL.execute_ c "DELETE FROM minting_policy_events"
+    pure $ ix & Core.dbLastSync .~ C.ChainPointAtGenesis
+  rollback p@(C.ChainPoint sno _) ix = do
+    let c = ix ^. Core.handle
+    liftIO $ SQL.execute c "DELETE FROM minting_policy_events WHERE slotNo > ?" (SQL.Only sno)
+    pure $ ix & Core.dbLastSync .~ p
```

Comment:
The more I see this code, the more I think we could provide helpers that takes a `[(TableName, SlotNoField)]` and generate the two rollback functions.